enhancements/cloud-integration/azure/azure-workload-identity.md

```diff
@@ -0,0 +1,368 @@
+
+### Goals
+
+- Core OpenShift operators utilize short-lived, bound service account token credentials to authenticate with Azure API Services.
+- Self-managed OpenShift administrators can create Azure infrastructure necessary to utilize workload identity federation such as an Azure blob storage container based OIDC using `ccoctl`.
+- Self-managed OpenShift administrators can create Azure Managed Identities via `ccoctl`'s processing of `CredentialsRequest` custom resources extracted from the release image prior to installation and provide the secrets output as manifests for installation which serve as the credentials for core OpenShift operators.
+- A user can utilize a Federated Azure Managed Identity Credential for their workload using the [mutating admission webhook provided by Azure Workload Identity](https://azure.github.io/azure-workload-identity/docs/installation/mutating-admission-webhook.html).
+
+### Non-Goals
+
+- Creation of Azure Managed Identity infrastructure (OIDC, managed identities, federated credentials) in managed environments (eg. ARO).
+- Role granularity for the explicit necessary permissions granted to
+  Managed Identities. Permissions needed by operator identities are
+  enumerated within `CredentialsRequests` for platforms such as AWS,
+  example:
+  [aws-ebs-csi-driver-operator](https://github.com/openshift/cluster-storage-operator/blob/f1ddb697afb3c33d6d45936e58fad101abe26f13/manifests/03_credentials_request_aws.yaml). Granular
+  permissions for operators on Azure are not a goal of this
+  enhancement but should be implemented either in parallel to this
+  enhancement or as a followup.
+
+## Proposal
+
+In this proposal, the Cloud Credential Operator's command-line utility (`ccoctl`) will be extended with subcommands for Azure which will provide methods for generating the Azure infrastructure (blob container OIDC, managed identities and federated credentials) and secret manifests necessary to create an Azure cluster that utilizes Azure Workload Identity for core OpenShift operator authentication.
+
+OpenShift operators as well as the Installer will be updated to create Azure clients using a bound `ServiceAccount` token that has been associated with a Managed Identity (identified by `clientID`) in Azure. Operators or repositories that we expect will need changes, listed in [CCO-235](https://issues.redhat.com/browse/CCO-235):
+
+- [Installer](https://github.com/openshift/installer)
+- [cloud-credential-operator](https://github.com/openshift/cloud-credential-operator)
+- [cluster-image-registry-operator](https://github.com/openshift/cluster-image-registry-operator)
+- [cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator)
+- [cluster-storage-operator](https://github.com/openshift/cluster-storage-operator)
+- [machine-api-operator](https://github.com/openshift/machine-api-operator)
```

Comment:
Have you checked `machine-api-provider-azure` and `cluster-api-provider-azure` support what they need to?

Reply:
@JoelSpeed `machine-api-provider-azure` was recently migrated from adal to azidentity in https://github.com/openshift/machine-api-provider-azure/pull/48 and we will need to add client changes around [here](https://github.com/openshift/machine-api-provider-azure/blob/3405d8c4b87fc22a5ac355cd4a9611a38aa1f3cc/pkg/cloud/azure/actuators/machine_scope.go#L334-L363) to pick up proposed secret field changes and create a client via `azidentity.NewClientAssertionCredential`. Adding to the list.

The `cluster-api-provider-azure` implementation is still in progress (https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2924) but looks compatible with our approach. We will at least need to bring these changes in so will add to the list.